pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -849,27 +864,49 @@ func Run() {
 		case <-stillRunning.C:
 		}
 		agentlog.StillRunning(agentName)
+
+		//if the trigger publish device info has failed, retry
+		retryTriggerPublishDevInfo(&zedagentCtx)
+	}
+}
+
+// if trigger publish device info has failed, try again
+func retryTriggerPublishDevInfo(ctxPtr *zedagentContext) {
+	if ctxPtr.devInfoTriggerFail && ctxPtr.devInfoTaskReady {
+		log.Infof("Retrying trigger PublishDeviceInfo\n")
+		ctxPtr.devInfoTriggerFail = false
+		triggerPublishDevInfo(ctxPtr)
 	}
 }
 
 func triggerPublishDevInfo(ctxPtr *zedagentContext) {
 
+	// last publish event has to be complete,
+	// for handling current publish event trigger,
+	// wait until then
+	if ctxPtr.devInfoTriggerFail {
+		return
+	}
 	log.Info("Triggered PublishDeviceInfo")
 	select {
 	case ctxPtr.TriggerDeviceInfo <- struct{}{}:
 		// Do nothing more
 	default:
-		log.Info("Failed to send on PublishDeviceInfo")
+		log.Errorf("Failed to send on PublishDeviceInfo")
```

Comment:
Where do we see the error as to why the publish failed? Would be good to print it here..

Reply:
The device is generating multiple events which end up in issuing triggers for publishDevInfo. This is good to keep a tab on the failures while issuing triggers for publistDevInfo.